File: trunk/IdentityGenerator/java/test/com/seovic/coherence/identity/SequenceGeneratorTests.java
package com.seovic.coherence.identity;


import com.seovic.coherence.identity.sequence.SequenceGenerator;
import com.seovic.coherence.identity.sequence.Sequence;

import com.tangosol.net.NamedCache;
import com.tangosol.net.CacheFactory;

import static org.junit.Assert.*;

import org.junit.Before;
import org.junit.Test;


/**
 * Tests for SequenceGenerator.
 *
 * @author  Aleksandar Seovic  2008.11.24
 */
public class SequenceGeneratorTests
    {
    private static NamedCache sequenceCache = CacheFactory.getCache("sequences");

    @Before
    public void before() throws Exception {
        sequenceCache.clear();
    }

    @Test
    public void testIdGeneration() {
        IdentityGeneratorClient igc = new IdentityGeneratorClient(new SequenceGenerator("test"));

        assertEquals(100, igc.generateIdentities(1, 100).size());
        assertEquals(100, igc.generateIdentities(5, 20).size());
        assertEquals(100, igc.generateIdentities(10, 10).size());

        Sequence seq = (Sequence) sequenceCache.get("test");
        assertEquals(300, seq.peek());
    }

    @Test
    public void testIdGenerationWithoutBlockCaching() {
        IdentityGeneratorClient igc = new IdentityGeneratorClient(new SequenceGenerator("test", 1));

        assertEquals(100, igc.generateIdentities(5, 20).size());

        Sequence seq = (Sequence) sequenceCache.get("test");
        assertEquals(100, seq.peek());
    }

    @Test
    public void testIdGenerationWithMultipleClients() {
        IdentityGeneratorClient igc1 = new IdentityGeneratorClient(new SequenceGenerator("test", 10));
        IdentityGeneratorClient igc2 = new IdentityGeneratorClient(new SequenceGenerator("test", 10));

        assertEquals(25, igc1.generateIdentities(5, 5).size());
        assertEquals(25, igc2.generateIdentities(5, 5).size());

        Sequence seq = (Sequence) sequenceCache.get("test");
        assertEquals(60, seq.peek());
    }

}

File: trunk/main.net/test/Coherence.Tools.Tests/Identity/SequenceGeneratorTests.cs
using System.Collections;
using NUnit.Framework;
using Seovic.Coherence.Identity.Sequence;
using Tangosol.Net;

namespace Seovic.Coherence.Identity
{
    [TestFixture]
    public class SequenceGeneratorTests
    {
        private static INamedCache sequenceCache =
            CacheFactory.GetCache(SequenceGenerator.CACHE_NAME);

        [SetUp]
        public void SetUp()
        {
            sequenceCache.Clear();
        }

        [Test]
        public void TestIdGeneration()
        {
            IdentityGeneratorClient<long> igc = new IdentityGeneratorClient<long>(
                new SequenceGenerator("test"));

            Assert.AreEqual(100, igc.GenerateIdentities(1, 100).Count);
            Assert.AreEqual(100, igc.GenerateIdentities(5, 20).Count);
            Assert.AreEqual(100, igc.GenerateIdentities(10, 10).Count);

            Sequence.Sequence seq = (Sequence.Sequence) sequenceCache["test"];
            Assert.AreEqual("test", seq.Name);
            Assert.AreEqual(300, seq.Peek);
        }

        [Test]
        public void TestIdGenerationWithoutBlockCaching()
        {
            IdentityGeneratorClient<long> igc = new IdentityGeneratorClient<long>(
                    new SequenceGenerator("test", 1));

            Assert.AreEqual(100, igc.GenerateIdentities(5, 20).Count);

            Sequence.Sequence seq = (Sequence.Sequence)sequenceCache["test"];
            Assert.AreEqual("test", seq.Name);
            Assert.AreEqual(100, seq.Peek);
        }

        [Test]
        public void TestIdGenerationWithMultipleClients()
        {
            IdentityGeneratorClient<long> igc1 = new IdentityGeneratorClient<long>(
                    new SequenceGenerator("test", 10));
            IdentityGeneratorClient<long> igc2 = new IdentityGeneratorClient<long>(
                    new SequenceGenerator("test", 10));

            Assert.AreEqual(25, igc1.GenerateIdentities(5, 5).Count);
            Assert.AreEqual(25, igc2.GenerateIdentities(5, 5).Count);

            Sequence.Sequence seq = (Sequence.Sequence)sequenceCache["test"];
            Assert.AreEqual("test", seq.Name);
            Assert.AreEqual(60, seq.Peek);
        }
    }
}
